core/templates/pages/topic-editor-page/services/topic-editor-state.service.ts

```diff
@@ -18,541 +18,556 @@
-            AlertsService.addWarning(
-              error || 'There was an error when saving the topic.');
-            _topicIsBeingSaved = false;
-          });
-        return true;
-      },
+          }
+          this.undoRedoService.clearChanges();
+          this._topicIsBeingSaved = false;
+          resolve();
+        }, (error: string) => {
+          this.alertService.addWarning(
+            error || 'There was an error when saving the topic.');
+          this._topicIsBeingSaved = false;
+        });
+    });
+  }
 
-      /**
-       * Returns whether this service is currently attempting to save the
-       * topic maintained by this service.
-       */
-      isSavingTopic: function() {
-        return _topicIsBeingSaved;
-      },
+  /**
+   * Returns whether this service is currently attempting to save the
+   * topic maintained by this service.
+   */
+  isSavingTopic(): boolean {
+    return this._topicIsBeingSaved;
+  }
 
-      get onTopicInitialized() {
-        return _topicInitializedEventEmitter;
-      },
+  get onTopicInitialized(): EventEmitter<void> {
+    return this._topicInitializedEventEmitter;
+  }
 
-      get onTopicReinitialized() {
-        return _topicReinitializedEventEmitter;
-      },
-      /**
+  get onTopicReinitialized(): EventEmitter<void> {
+    return this._topicReinitializedEventEmitter;
+  }
+  /**
        * Returns the classroom name for the topic.
        */
-      getClassroomUrlFragment: function() {
-        return _classroomUrlFragment;
-      },
+  getClassroomUrlFragment(): string {
+    return this._classroomUrlFragment;
+  }
 
-      /**
-       * Attempts to set the boolean variable _topicWithNameExists based
-       * on the value returned by doesTopicWithNameExistAsync and
-       * executes the success callback provided. No arguments are passed to the
-       * success callback. Execution of the success callback indicates that the
-       * async backend call was successful and that _topicWithNameExists
-       * has been successfully updated.
-       */
-      updateExistenceOfTopicName: function(topicName, successCallback) {
-        EditableTopicBackendApiService.doesTopicWithNameExistAsync(
-          topicName).then(
-          function(topicNameExists) {
-            _setTopicWithNameExists(topicNameExists);
-            if (successCallback) {
-              successCallback();
-            }
-            $rootScope.$applyAsync();
-          }, function(error) {
-            AlertsService.addWarning(
-              error ||
+  /**
+   * Attempts to set the boolean variable _topicWithNameExists based
+   * on the value returned by doesTopicWithNameExistAsync and
+   * executes the success callback provided. No arguments are passed to the
+   * success callback. Execution of the success callback indicates that the
+   * async backend call was successful and that _topicWithNameExists
+   * has been successfully updated.
+   */
+  updateExistenceOfTopicName(
```

Comment:
This does an async operation and therefore we had used a `$rootScope.$applyAsync();`. but that has been removed. How are we handling this now?

Reply:
Made the function to return a promise. Every angularJS file where this function was called, I have added an $rootScope.$applyAsync();. This removes the requirement of using $rootScope.$applyAsync(); in the function itself